src/cargo/core/compiler/compilation.rs

```diff
@@ -356,6 +356,7 @@ impl<'gctx> Compilation<'gctx> {
         // in BuildContext::target_metadata()
         let rust_version = pkg.rust_version().as_ref().map(ToString::to_string);
         cmd.env("CARGO_MANIFEST_DIR", pkg.root())
+            .env("CARGO_MANIFEST_PATH", pkg.manifest_path())
```

Comment:
This change needs an FCP, and maybe an unstable gate before `-Zscript` stabilization.

Reply:
Resolved by <https://github.com/rust-lang/cargo/pull/14404#issuecomment-2292019184> 😁.

Reply:
I figure this is small enough in scope and likely useful enough outside of cargo-script that we could stabilize now, with an FCP, independent of the cargo-script progress.